README.ideal.md

```diff
@@ -0,0 +1,145 @@
+Ni-Mh (called “nickel-nix” or “Nixel” at the moment, but I really like Ni-Mh) aims at being your main entrypoint for managaing all these different tools, so that you can get:
+
+1. A unified configuration framework for all of these
+2. A powerful and ergonomic language to allow you to easily abstract over these configurations, with discoverability and early error reporting
+
+## Getting started
+
+To start using `nickel-nix`, you need `Nix` to be installed and configured.
+If it isn't already the case, you can get it with:
+
+```console
+$ curl -L https://nixos.org/nix/install | bash
+# We also need a couple of experimental Nix features
+$ mkdir -p ~/.config/nix
+$ echo 'experimental-features = nix-command flakes' >> ~/.config/nix/nix.conf
+```
+
+Then bootstrap a project with:
+
+```console
+$ nix flake init -t github:nickel-lang/nickel-nix
+# Edit the project file to fit your needs
+$ $EDITOR project.ncl
+# Enter the environment
+$ nix develop
+```
+
+## Managing your environment with Nixel
+
+### Dependencies
+
+Nixel can be used to declare the dependencies for your project.
+These can then be instantiated using [Nix](https://nixos.org/nix).
+
+> TODO: Document
+
+### CI configuration
+
+You can use Nixel to generate a CI configuration.
+This is extremely handy if you want to test some non-trivial matrix of platforms/configuration.
+
+Only Github actions is supported at the moment, but we plan to add more options soon.
+
+> TODO: Document
+
+### Development services
+
+Nixel can also be used for some simple service management.
+In combination with the Nix integration, this allows you to quickly spawn any service that you might depend on.
+
+Once the services have been defined in your `project.ncl` file, you can run `nixel up` to start them.
+
+> TODO: Document
+
+### Deployment
+
+If you want to push things yet a bit further, Nixel can also interoperate with [tf-ncl](https://github.com/tweag/tf-ncl) to allow you to manage the deployment infrastructure for your project.
+
+> TODO: Document
+
+## Structure of the `project.ncl` file
+
+`project.ncl` is the main entrypoint for `nickel-nix`.
+It describes the packages that should be included in your development environment, as well as the template for any file you'd like to generate:
+
+```nickel
+let Nixel = import ".nickel-nix/lock.ncl" in
+let from_nixpkgs = Nixel.nix.from_nixpkgs in
+{
+  shells = Nixel.shells.Rust,
+  shells = Nixel.shells.Nickel,
+
+  shells.build = {
+    packages = [from_nixpkgs "pandoc"],
+    scripts = {
+      build = nix-s%"
+                #!/usr/bin/env bash
+
+                %{from_nixpkgs "gnumake"}/bin/make -j$(nproc) -l$(nproc)
+            "%,
+    },
```

Comment:
Do we really need `scripts` as a separate field? This feels like a bad abstraction that hides internals from the user without much improvement in ergonomics. Why not just `packages = [ <...> make_bash_script nix-s%"${from_nixpkgs "gnumake"}/bin/make <...>"% ];` ?

Reply:
We don't strictly need this indeed. I think it makes sense to have it as a lightweight special-case though (the same way you can have for instance `systemd.services.foo.script = "echo foo";` and `systemd.services.foo.ServiceConfig.ExecStart = pkgs.writeScript "foo" "echo foo";`). Also, we could build some cool stuff on top of that – either exposing these in a special way from the CLI, or adding a bit of documentation for them that could be exposed to the user.